### PR TITLE
Update docker compose to use healthchecks to avoid race conditions

### PR DESCRIPTION
**Pull request by RyanMacG**

### CODE REVIEW

**Comment by RyanMacG** [comment]
Requires docker to be upgraded first

File: docker-compose.yml [hunk 1]
```diff
@@ -35,6 +42,11 @@ services:
       - http.host=0.0.0.0
       - transport.host=127.0.0.1
       - xpack.security.enabled=false
+    healthcheck:
+      test: ["CMD-SHELL", "curl --silent --fail localhost:9200/_cluster/health || exit 1"]
```
Comment (adrianclay) [inline docker-compose.yml]: Be nice to standardise on either using string format or the "CMD-SHELL" format.

File: docker-compose.yml [hunk 2]
```diff
@@ -35,6 +43,11 @@ services:
       - http.host=0.0.0.0
       - transport.host=127.0.0.1
       - xpack.security.enabled=false
+    healthcheck:
+      test: ["CMD", "curl --silent --fail localhost:9200/_cluster/health || exit 1"]
+      interval: 30s
+      timeout: 30s
+      retries: 3
```
Comment (kevindew) [inline docker-compose.yml]: These seem somewhat magical. Can you include in commit or similar some info on why these values were chosen and why this case needs them and others are ok with the defaults?

**Comment by RyanMacG** [comment]
The most recent commit ec5c4b83  covers both the race condition we ran into where the collections container was live but not healthy and hopefully any other applications being in a similar state.

**Comment by thomasleese** [comment]
I noticed here you're adding the health checks into the `docker-compose.yml` file.

I wonder if it's better to [include them directly into the `Dockerfile`s for each app](https://docs.docker.com/engine/reference/builder/#healthcheck)? It means then the maintainers of the apps can keep them up to date, and if we decide to use Docker for something other than the end to end tests, we get those health checks automatically.

**Pull request merged**